lib/statesman/adapters/active_record_queries.rb

```diff
@@ -30,6 +34,30 @@ def most_recent_transition_join
 
         private
 
+        def in_state_without_initial(*states)
+          states = states.flatten.map(&:to_s)
+
+          joins(most_recent_transition_join_without_initial).
+            where(states_where(most_recent_transition_alias, states), states).
+            where("#{most_recent_transition_alias}.to_state != ?", initial_state)
+        end
+
+        def not_in_state_without_initial(*states)
+          states = states.flatten.map(&:to_s)
+
+          joins(most_recent_transition_join_without_initial).
+            where("NOT (#{states_where(most_recent_transition_alias, states)})",
+                  states).
+            where("#{most_recent_transition_alias}.to_state != ?", initial_state)
+        end
+
+        def most_recent_transition_join_without_initial
+          "JOIN #{model_table} AS #{most_recent_transition_alias}
```

Comment:
The meat of this method and `most_recent_transition_join` are the same apart from the `JOIN` condition at the beginning. Do you think it would make sense to have

```
def most_recent_transition_join
  "INNER JOIN " + join_statement
end

def most_recent_transition_join_without_initial
  "JOIN " + join_statement
end

private

def join_statement
  "#{model_table} AS #{most_recent_transition_alias}
     ON #{table_name}.id = #{most_recent_transition_alias}.#{model_foreign_key}	                  
     AND #{most_recent_transition_alias}.most_recent = #{db_true}"
end
```

to ensure they are always update in-sync.

Reply:
Good call, I fixed it in https://github.com/gocardless/statesman/pull/343/commits/5139e5d7ddf5a1139d85c6d12d234c5f9085c411, preserving the existing `LEFT OUTER JOIN` in the `most_recent_transition_join` method.